virttest/qemu_vm.py

```diff
@@ -1645,11 +1652,22 @@ def sort_key(dev):
         vcpu_threads = params.get_numeric("vcpu_threads")
         vcpu_dies = params.get("vcpu_dies", 0)
         enable_dies = vcpu_dies != "INVALID" and Flags.SMP_DIES in devices.caps
-        if not enable_dies:
+        vcpu_clusters = params.get("vcpu_clusters", 0)
+        enable_clusters = vcpu_clusters != "INVALID" and Flags.SMP_CLUSTERS in devices.caps
+
+        if enable_dies and int(vcpu_dies) > 0:
+            # PC target support SMP 'dies' parameter since qemu 4.1
+            vcpu_dies = int(vcpu_dies)
+        else:
```

Comment:
```suggestion
        elif not enable_dies:
```
The condition `int(vcpu_dies) == 0` is also belongs to the condition `else:` block, so it can not follow the previous condition` if not enable_dies:` for setting `vcpu_dies = 1`

Reply:
This is the expected setting, because the algorithm below has changed so it needs to be reassigned to 1 here when `int(vcpu_dies) == 0`.

Reply:
After discussion, keep the original design.
The case of `int(vcpu_dies) == 0` is handled separately below.